.idea/Working version/algorithms.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * Created by tsodergren on 6/5/17.
+ */
+
+
+function constructEdges() {
+
+
+    var radiusChange = arguments.length ? arguments[0] : 0;
+
+    //squared coverage diameter and annulus of uncertainty
+    var sqDiameter = 4 * Math.pow(complexRadius, 2);
+    var sqDiameterMin = 4 * Math.pow(complexRadius-dataRadius, 2);
+    var sqDiameterMax = 4 * Math.pow(complexRadius+dataRadius, 2);
+
+
+    if (!a_edges) {
+        var x=math.zeros(numSamples, numSamples,'sparse');
+        a_edges = {edgeProb: x.clone(), edgeAnchorDist: x.clone(), edgeDist: x.clone(), edgeFlag: x.clone(),
+            edgeInd: x.clone() };
+    }
+
+    var iEdgeDist, iEdgeFlag, tempEdges;
+
+    if ( radiusChange==1 ) {
+        tempEdges = JSON.parse(JSON.stringify(ripsEdges));
```

Comment:
Why do you need to do this? When you load in the file with d3.json it already formats it into an array.